cmdx/ref.go

```diff
@@ -15,25 +15,36 @@ import (
 // This should be added on the root command and can
 // be used as `help reference` or `reference help`.
 func SetRefCmd(root *cobra.Command) *cobra.Command {
+	var stylized bool
 	cmd := &cobra.Command{
 		Use:   "reference",
 		Short: "Comprehensive reference of all commands",
 		Long:  referenceLong(root),
-		Run:   referenceHelpFn(),
+		Run:   referenceHelpFn(&stylized),
 		Annotations: map[string]string{
 			"group": "help",
 		},
 	}
-	cmd.SetHelpFunc(referenceHelpFn())
+	cmd.SetHelpFunc(referenceHelpFn(&stylized))
+	cmd.Flags().BoolVarP(&stylized, "stylized", "s", true, "output with ansi color if true")
```

Comment:
@mabdh I think we should keep color enabled by default. And if standard NO_COlOR is set then we should return plain. Which i think markdown printer already takes care of. So we might not need this bool.

Reply:
@ravisuhag that works great only if we want to show the reference in terminal. If we want to write the reference to a `.md` file, all ansi color characters also written down. Even if we stripped up the ansi color chars, bacticks are gone so the markdown does not rendered properly.

example guardian
```shell
 go run . reference > ./docs/docs/reference/cli.md
```

<img width="1056" alt="Screen Shot 2022-10-20 at 07 13 43" src="https://user-images.githubusercontent.com/4079709/196827621-7adf339d-c5cb-4961-8f9f-d19104de68fa.png">

I think this would be great if `reference` command is reusable (could be rendered with color or just plain markdown).